source/Calamari.Shared/Integration/Certificates/WindowsCertificateStoreLocator.cs

```diff
@@ -0,0 +1,101 @@
+                }
+
+                store.Close();
+            }
+
+            return null;
+        }
+        
+        public static ICollection<string> GetStoreNames(StoreLocation location)
+        {
+            lock (StoreNamesSyncLock)
+            {
+                var callback = new CertEnumSystemStoreCallBackProto(CertEnumSystemStoreCallBack);
+                var names = new List<string>();
+                CertificateSystemStoreLocation locationFlags;
+
+                switch (location)
+                {
+                    case StoreLocation.CurrentUser:
+                        locationFlags = CertificateSystemStoreLocation.CurrentUser;
+                        break;
+                    case StoreLocation.LocalMachine:
+                        locationFlags = CertificateSystemStoreLocation.LocalMachine;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(location), location, null);
+                }
+
+                EnumeratedStoreNames.Clear();
+                CertEnumSystemStore(locationFlags, IntPtr.Zero, IntPtr.Zero, callback);
+                names.AddRange(EnumeratedStoreNames);
+
+                return names;
+            }
+        }
+        
+        /// <summary>
+        /// call back function used by CertEnumSystemStore
+        ///
+        /// Currently, there is no managed support for enumerating store
+        /// names on a machine. We use the win32 function CertEnumSystemStore()
+        /// to get a list of stores for a given context.
+        ///
+        /// Each time this callback is called, we add the passed store name
+        /// to the list of stores
+        /// </summary>
+        static bool CertEnumSystemStoreCallBack(string storeName, uint dwFlagsNotUsed, IntPtr notUsed1, IntPtr notUsed2, IntPtr notUsed3)
+        {
+            EnumeratedStoreNames.Add(storeName);
+            return true;
+        }
+        
+        [DllImport("Crypt32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
+        static extern
+            bool CertEnumSystemStore(CertificateSystemStoreLocation dwFlags, IntPtr notUsed1, IntPtr notUsed2,
+                                     CertEnumSystemStoreCallBackProto fn);
+        
+        enum CertificateSystemStoreLocation
+        {
+            CurrentUser = 1 << 16, // CERT_SYSTEM_STORE_CURRENT_USER
+            LocalMachine = 2 << 16, // CERT_SYSTEM_STORE_LOCAL_MACHINE
+            CurrentService = 4 << 16, // CERT_SYSTEM_STORE_CURRENT_SERVICE
+            Services = 5 << 16, // CERT_SYSTEM_STORE_SERVICES
+            Users = 6 << 16, // CERT_SYSTEM_STORE_USERS
+            UserGroupPolicy = 7 << 16, // CERT_SYSTEM_STORE_CURRENT_USER_GROUP_POLICY
+            MachineGroupPolicy = 8 << 16, // CERT_SYSTEM_STORE_LOCAL_MACHINE_GROUP_POLICY
+            LocalMachineEnterprise = 9 << 16, // CERT_SYSTEM_STORE_LOCAL_MACHINE_ENTERPRISE
+        }
```

Comment:
This looks like it's been copied from somewhere, but we haven't removed it in the PR?

Reply:
:D I noticed that when I just looked at the PR before and wondered if you would pick it up before I got around to making an update.
Good catch